timelock-agent/src/main/java/com/palantir/timelock/paxos/AsyncTimeLockServicesCreator.java

```diff
@@ -119,7 +125,19 @@ private AsyncTimelockService createRawAsyncTimelockService(
         return AtlasDbMetrics.instrumentWithTaggedMetrics(
                 metricRegistry, serviceClass, service, MetricRegistry.name(serviceClass),
                 context -> ImmutableMap.of(
-                        "client", client,
-                        "isCurrentSuspectedLeader", String.valueOf(leadershipCreator.isCurrentSuspectedLeader())));
+                        AtlasDbMetricNames.TAG_CLIENT, client,
+                        AtlasDbMetricNames.TAG_CURRENT_SUSPECTED_LEADER, String.valueOf(
+                                leadershipCreator.isCurrentSuspectedLeader())));
+    }
+
+    private void deregisterLeaderMetrics() {
```

Comment:
I think this should be pushed down into the `MetricsManager` class, it stores an additional list of metrics which should be kept up to date.  Have a `deregisterTaggedMetrics(Predicate<MetricName> filterCondition)` method.

Reply:
done!